homeassistant/components/zabbix/__init__.py

```diff
@@ -68,101 +97,292 @@
+                filter_dict[filter_key] = hass_data[INCLUDE_EXCLUDE_FILTER].get(
+                    filter_key
+                )
+            else:
+                filter_dict[filter_key] = []
+        entities_filter = convert_filter(filter_dict)
 
-    def event_to_metrics(
-        event: Event, float_keys: set[str], string_keys: set[str]
-    ) -> list[ZabbixMetric] | None:
-        """Add an event to the outgoing Zabbix list."""
-        state = event.data.get("new_state")
-        if state is None or state.state in (STATE_UNKNOWN, "", STATE_UNAVAILABLE):
-            return None
+        zabbix_sender = await hass.async_add_executor_job(
+            lambda: Sender(server=host, port=port)
+        )
 
-        entity_id = state.entity_id
-        if not entities_filter(entity_id):
-            return None
+        hass.data[DOMAIN][entry.entry_id][ZABBIX_SENDER] = zabbix_sender
+        hass.data[DOMAIN][entry.entry_id][ENTITIES_FILTER] = entities_filter
+        instance = ZabbixThread(hass, entry.entry_id)
+        hass.data[DOMAIN][entry.entry_id][ZABBIX_THREAD_INSTANCE] = instance
+        await hass.async_add_executor_job(instance.setup, hass)
+        _LOGGER.info(
+            "Started Zabbix thread for sharing events to zabbix_sender for config entry %s:",
+            entry.entry_id,
+        )
 
-        floats = {}
-        strings = {}
+    if hass_data.get(CONF_USE_API, False) and hass_data.get(CONF_USE_SENSORS, False):
+        # define Zabbix API for sensors
         try:
-            _state_as_value = float(state.state)
-            floats[entity_id] = _state_as_value
-        except ValueError:
-            try:
-                _state_as_value = float(state_helper.state_as_number(state))
-                floats[entity_id] = _state_as_value
-            except ValueError:
-                strings[entity_id] = state.state
-
-        for key, value in state.attributes.items():
-            # For each value we try to cast it as float
-            # But if we cannot do it we store the value
-            # as string
-            attribute_id = f"{entity_id}/{key}"
-            try:
-                float_value = float(value)
-            except (ValueError, TypeError):
-                float_value = None
-            if float_value is None or not math.isfinite(float_value):
-                strings[attribute_id] = str(value)
-            else:
-                floats[attribute_id] = float_value
-
-        metrics = []
-        float_keys_count = len(float_keys)
-        float_keys.update(floats)
-        if len(float_keys) != float_keys_count:
-            floats_discovery = [{"{#KEY}": float_key} for float_key in float_keys]
-            metric = ZabbixMetric(
-                publish_states_host,
-                "homeassistant.floats_discovery",
-                json.dumps(floats_discovery),
+            zapi = await hass.async_add_executor_job(
```

Comment:
Let's put everything in one function and just use one executor thread

Reply:
Done.
Also same used for config_flow.py (_create_template_job function) as well.